### ISSUE TITLE
Ability to integrate Jasonette as framework in exiting app

### ISSUE THREAD

**Issue by anivaros** [Jasonette/JASONETTE-iOS#223]
It would be great if developers will get ability to use Jasonette as framework, not as app.

For example, init some component by passing template and layout to initializer, or something like that.

**Comment by gtramontina** [comment]
I believe this is already doable. Have a look at https://docs.jasonette.com/advanced/ (specially https://docs.jasonette.com/advanced/#integration) and let us know if that addresses your suggestion.

**Comment by anivaros** [comment]
Yes, I’ve read this article, but in current implementation I should copy code into my project and loose support of future updates in this repo. It would be cool if I could connect jasonnete as “pod”, for example, and stay in touch with code updates.

**Comment by gtramontina** [comment]
I see… I'm not sure if turning Jasonette into a pod is in the roadmap (@gliechtenstein?). Would using it as a git submodule (as opposed to copying) help your concerns re: getting the latest updates?

**Comment by anivaros** [comment]
Maybe yes, I'll think about that, thanks for an idea.
Btw, if I'll try to fork and implement my suggestion, is there any chance it will be merged in this repo?

**Comment by gliechtenstein** [comment]
@anivaros Yes i am a total fan of the idea, it sounds like a great idea! I just haven't had time to work on something like that yet because of time/resource constraints.

Let me share some of my thoughts on this, just to make sure we're on the same page:

1. We actually have a project called [JEM (Jason Extension Manager)](https://github.com/Jasonette/JEM) which is almost complete but was put on hold because other higher priority things came up. (But I plan on getting back to it soon, once I finally release Jasonette-Web)
2. JEM is an electron based app that makes it easy to register, share and install Jasonette extensions into an **existing Jasonette app** (not precisely what we're talking about here). You can think of it as sort of a Jasonette version of cocoapods.
3. The current version of JEM only supports actions, because actions were more urgent than UI components, but the plan is to support components as well.
4. You may ask "Why not just use CocoaPods? Why re-invent the wheel?". Here's the answer: Our initial attempt was in fact using CocoaPods (for iOS): https://github.com/Jasonette/JASONETTE-iOS/issues/26
5. But the CocoaPods approach proved to be very cumbersome because you have to jump through all the hoops of learning to use CocoaPods, creating a PodSpec, submitting a PR so it can be registered to the main CocoaPods repo, etc. when all we really need is a single pair of `.h` and `.m` files that are completely **state-free** and **dependency-free** (Meaning one extension doesn't depend on another extension) Furthermore, this inconvenience was 100 times worse when it came to Android side, when we tried to use gradle (and gradle is supposed to be the simplest way to manage libraries on Android). Lastly, building a cross-platform unified UI for managing these extensions became really difficult because we somehow needed to create a wrapper that communicates with both CocoaPods and Gradle in a uniform manner.
6. So we had some discussions on this https://github.com/Jasonette/JASONETTE-iOS/issues/107 and decided that Jasonette has a specific enough approach that we can come up with a more focused and simple solution, which was to build our own workflow full stack. And that's how we came to build JEM. Please feel free to play with [JEM](https://github.com/Jasonette/JEM) it's not polished but definitely functional. Plus it works for both iOS and Android.

That said, your original question was how to use these components for other **non-Jasonette** projects, so I do realize these are not directly related. 

The reason I went through all the details was because we **do** want to modularize the UI components somehow, **BUT** I think it would be great if we could use the approach taken by JEM instead of publishing each component as cocoapod.

Let me know what you think. Would love to hear your idea.

**Comment by magnett** [comment]
Making it a framework or a cocoapod would be great but i understand all the points you made in your last post. You could however post a detailed tutorial or even better a video showing how-to "...copy all  Jasonette files into your own project..." because it sounds really challenging adding your (Jasonette) workspace into another/existing workspace in Xcode. So, to summarise, I would be really glad to read/watch a walkthrough where an existing single-view application with a single cocoapod (just to make it an actual workspace) could use Jasonette. I guess that if this approach is actually working, you could make up a video in no time.
Thanks in advance.
